include/psen_scan_v2/diagnostics.h

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2020 Pilz GmbH & Co. KG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#ifndef PSEN_SCAN_V2_DIAGNOSTICS_H
+#define PSEN_SCAN_V2_DIAGNOSTICS_H
+
+#include <array>
+#include <string>
+#include <map>
+#include <vector>
+#include <set>
+
+#include <fmt/format.h>
+#include <fmt/ostream.h>
+
+#include "psen_scan_v2/scanner_ids.h"
+
+namespace psen_scan_v2
+{
+static constexpr uint32_t DIAGNOSTIC_MESSAGE_RAW_LENGTH_FOR_ONE_DEVICE_IN_BYTES{ 9 };
+static constexpr uint32_t DIAGNOSTIC_MESSAGE_UNUSED_OFFSET_IN_BYTES{ 4 };
+static constexpr uint32_t DIAGNOSTIC_DATA_LENGTH_IN_BYTES{ DIAGNOSTIC_MESSAGE_UNUSED_OFFSET_IN_BYTES +
```

Comment:
Why do we write something like:
```
struct MonitoringFrameAdditionalFieldHeaderIds
{
  static constexpr MonitoringFrameAdditionalFieldHeaderId SCAN_COUNTER{ 0x02 };
  static constexpr MonitoringFrameAdditionalFieldHeaderId MEASURES{ 0x05 };
  static constexpr MonitoringFrameAdditionalFieldHeaderId END_OF_FRAME{ 0x09 };
  static constexpr MonitoringFrameAdditionalFieldHeaderId DIAGNOSTICS{ 0x04 };
};
```
But here we simply write everything without explicit binding via `struct` or something. I think both versions are not very good. I suggest using namespaces, in both cases.

Reply:
On it